File: Server/src/main/content/global/skill/production/cooking/recipe/potato/impl/MushroomPotato.java
package content.global.skill.production.cooking.recipe.potato.impl;

import content.global.skill.production.cooking.recipe.potato.PotatoRecipe;
import core.game.node.item.Item;

/**
 * The Mushroom potato.
 */
public class MushroomPotato extends PotatoRecipe {

    private static final Item MUSHROOM_POTATO = new Item(7058);

    private static final Item TOPPING = new Item(7066);

    @Override
    public Item getProduct() {
        return MUSHROOM_POTATO;
    }

    @Override
    public Item[] getIngredients() {
        return new Item[]{TOPPING};
    }

    @Override
    public boolean isTopping() {
        return true;
    }

    @Override
    public int getLevel() {
        return 64;
    }

    @Override
    public double getExperience() {
        return 10;
    }

}

File: Emulator/src/main/content/global/skill/cooking/type/potato/impl/MushroomPotato.kt
package content.global.skill.cooking.type.potato.impl

import content.global.skill.cooking.type.potato.PotatoRecipe
import org.rs.consts.Items
import core.game.node.item.Item

/**
 * Mushroom potato.
 */
class MushroomPotato : PotatoRecipe() {

    override fun getProduct(): Item {
        return MUSHROOM_POTATO
    }

    override fun getIngredients(): Array<Item> {
        return arrayOf(TOPPING)
    }

    override fun isTopping(): Boolean {
        return true
    }

    override fun getLevel(): Int {
        return 64
    }

    override fun getExperience(): Double {
        return 10.0
    }

    companion object {
        private val MUSHROOM_POTATO = Item(Items.MUSHROOM_POTATO_7058)
        private val TOPPING = Item(Items.MUSHROOM_AND_ONION_7066)
    }
}
